src/Microsoft.PowerShell.Commands.Management/commands/management/Service.cs

```diff
@@ -440,7 +365,8 @@ private List<ServiceController> MatchingServicesByServiceName()
                 }
                 else
                 {
-                    foreach (ServiceController service in OneService(pattern))
+                    ServiceController service = GetOneService(pattern);
+                    if (service != null)
```

Comment:
It  seems `GetOneService() `never return null.

Reply:
When the service doesn't exist, accessing `sc.Status;` will throw `InvalidOperationException`. In that case the method will catch the exception and return null.

Reply:
Thanks!

Closed.